### PR TITLE
added flash message and modal

### PR DESCRIPTION
**Pull request by Hakdoff**
Overview
Added flash message
Change post details to modal
Put the forms in a separate files

Snippet
image
image
image

### CODE REVIEW

File: src/public/css/post.css [hunk 1]
```diff
@@ -0,0 +1,62 @@
+  body {
+        overflow: auto; /* Ensure body can scroll */
```
Comment (hipe01mark) [inline src/public/css/post.css]: is this comment necessary?

File: src/public/css/post.css [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        position: fixed; /* Stay in place */
+        z-index: 1; /* Sit on top */
+        left: 0;
+        top: 0;
+        width: 100%; /* Full width */
+        height: 100%; /* Full height */
+        overflow: auto; /* Enable scroll if needed */
+        background-color: rgb(0,0,0); /* Fallback color */
+        background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
+        padding-top: 60px;
```
Comment (hipe01mark) [inline src/public/css/post.css]: are these comments necessary?

File: src/public/css/post.css [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        font-size: 28px;
+        font-weight: bold;
+    }
+
+    .close:hover,
+    .close:focus {
+        color: black;
+        text-decoration: none;
+        cursor: pointer;
+    }
```
Comment (hipe01mark) [inline src/public/css/post.css]: break line!

File: src/resources/js/post.js [hunk 4]
```diff
@@ -44,11 +79,28 @@ document.addEventListener('DOMContentLoaded', function() {
...
+    }
+
+    // Like button color change
+    const likeButtons = document.querySelectorAll('.like-button');
+    likeButtons.forEach(button => {
+        if (button.getAttribute('data-liked') === 'true') {
+            button.style.color = 'red';
+        }
+    });
+});
```
Comment (hipe01mark) [inline src/resources/js/post.js]: break line!

File: src/resources/views/pages/dashboard/post-footer.blade.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                </div>
+                <div id="comment{{ $post->id }}"
+                    class="d-dashboard__border-gray mb-2 flex p-2 items-center w-full hidden transition-all duration-300">
+                    <div class="flex flex-col h-max-64 w-full">
+                        <div>
+                            <!-- Add Comment Form -->
+                            @include('pages.forms.add-comment-form')
+                        </div>
+                    </div>
+                </div>
```
Comment (hipe01mark) [inline src/resources/views/pages/dashboard/post-footer.blade.php]: break line!

File: src/resources/views/pages/forms/add-comment-form.blade.php [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+                                <label for="comment" class="w-full">Comment as <span
+                                        class="font-semibold">{{ auth()->user()->username }}</span></label>
+                                <div class="flex flex-end items-center gap-2">
+                                    <textarea class="d-dashboard__border-gray w-full mt-2 p-2 rounded-lg" name="content"
+                                        placeholder="Enter your comment here"></textarea>
+                                    <button
+                                        class="py-5 px-10 me-2 mt-1 text-sm font-medium text-gray-100 focus:outline-none bg-mygray rounded-lg border border-mygray focus:z-10 focus:ring-4 focus:ring-gray-100 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-100 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700"
+                                        type="submit">Comment</button>
+                                </div>
+                            </form>
```
Comment (hipe01mark) [inline src/resources/views/pages/forms/add-comment-form.blade.php]: breakline!

File: src/resources/views/pages/forms/add-post-form.blade.php [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+                    <textarea id="auto-resize-textarea" class="d-dashboard__write-post mb-3" name='content'></textarea>
+                    <div id="image-preview-container" class="flex justify-center hidden"></div>
+                </div>
+                <div class="d-dashboard__imageicon">
+                    <input type="file" id="image-upload" name="images[]" multiple accept="image/*"
+                        style="display: none;">@include('svg.image-upload') <button id="shareButton"
+                        class="d-dashboard__post-btn">Post</button>
+                </div>
+            </div>
+        </form>
```
Comment (hipe01mark) [inline src/resources/views/pages/forms/add-post-form.blade.php]: breakline!

File: src/resources/views/pages/forms/delete-comment-form.blade.php [hunk 8]
```diff
@@ -0,0 +1,8 @@
+ <form
+                                                    action="{{ route('delete.comment', ['comment' => $comment->id]) }}"
+                                                    method="POST" class="m-0 inline comment-delete-form "
+                                                    onsubmit="return confirmDeleteComment()">
+                                                    @csrf
+                                                    @method('DELETE')
+                                                    <button class="text-sm font-semibold ml-1">Delete</button>
+                                                </form>
```
Comment (hipe01mark) [inline src/resources/views/pages/forms/delete-comment-form.blade.php]: breakline!

File: src/resources/views/pages/forms/delete-post-form.blade.php [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<form action="{{ route('delete.post', ['post' => $post]) }}" method="post"
+                                    onsubmit="return confirmDeletePost()">
+                                    @csrf
+                                    @method('DELETE')
+                                    <button>
+                                        <span class="underline">Delete</span>
+                                    </button>
+                                </form>
```
Comment (hipe01mark) [inline src/resources/views/pages/forms/delete-post-form.blade.php]: breakline!

File: src/resources/views/pages/forms/edit-comment-form.blade.php [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+                                            action="{{ route('posts.comments.edit', ['comment' => $comment->id]) }}"
+                                            method="POST" class="hidden mt-1 mb-1 edit-form">
+                                            @csrf
+                                            @method('PUT')
+                                            <input type="text" name="content" value="{{ $comment->content }}"
+                                                class="border rounded w-full text-sm">
+                                            <button type="submit" class="text-sm font-semibold mt-1">Save</button>
+                                            <button type="button" class="text-sm font-semibold mt-1 ml-2"
+                                                onclick="cancelEdit({{ $comment->id }})">Cancel</button>
+                                        </form>
```
Comment (hipe01mark) [inline src/resources/views/pages/forms/edit-comment-form.blade.php]: breakline!

File: src/resources/views/pages/forms/edit-post-form.blade.php [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+            <form id="editPostForm{{ $post->id }}" action="/edit-post/{{ $post->id }}" method="POST"
+                class="hidden mt-1 mb-1 edit-form">
+                @csrf
+                @method('PUT')
+                <input type="text" name="content" value="{{ $post->content }}"
+                    class="border rounded w-full h-14 text-lg">
+                <button type="submit" class="text-sm font-semibold mt-1">Save</button>
+                <button type="button" class="text-sm font-semibold mt-1 ml-2"
+                    onclick="cancelEditPost({{ $post->id }})">Cancel</button>
+            </form>
```
Comment (hipe01mark) [inline src/resources/views/pages/forms/edit-post-form.blade.php]: breakline!

File: src/resources/views/pages/forms/share-post-form.blade.php [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+        <div class="mt-2">
+            <textarea id="auto-resize-textarea" class="d-dashboard__write-post mb-3" name='content'></textarea>
+            <div id="image-preview-container" class="flex justify-center hidden"></div>
+        </div>
+        <input type="hidden" name="post_id" id="sharePostId">
+        <div class="flex justify-end">
+            <button id="shareButton" class="d-dashboard__post-btn">Share</button>
+        </div>
+    </div>
+</form>
```
Comment (hipe01mark) [inline src/resources/views/pages/forms/share-post-form.blade.php]: breakline!

File: src/public/css/post.css (hunk 2 above)
Comment (hipe01mark) [reply src/public/css/post.css]: remove all comments

File: src/public/css/post.css [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+        display: none; 
+        position: fixed; 
+        z-index: 1; 
+        left: 0;
+        top: 0;
+        width: 100%; 
+        height: 100%; 
+        overflow: auto; 
+        background-color: rgb(0,0,0); 
+        background-color: rgba(0,0,0,0.4);
```
Comment (mikco-hipe) [inline src/public/css/post.css]: 2 background-color?

**Review by mikco-hipe — COMMENTED** [review]
Please check!

File: src/public/css/post.css [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+        font-weight: bold;
+    }
+
+    .close:hover,
+    .close:focus {
+        color: black;
+        text-decoration: none;
+        cursor: pointer;
+    }
+
```
Comment (hipe01mark) [inline src/public/css/post.css]: Breakline!

**Review by mikco-hipe — APPROVED** [review]
LTFRB 🙇

**Pull request merged**